clang/docs/analyzer/user-docs/TaintAnalysisConfiguration.rst

```diff
@@ -2,10 +2,13 @@
 Taint Analysis Configuration
 ============================
 
-The Clang Static Analyzer uses taint analysis to detect security-related issues in code.
-The backbone of taint analysis in the Clang SA is the `GenericTaintChecker`, which the user can access via the :ref:`alpha-security-taint-TaintPropagation` checker alias and this checker has a default taint-related configuration.
-The built-in default settings are defined in code, and they are always in effect once the checker is enabled, either directly or via the alias.
-The checker also provides a configuration interface for extending the default settings by providing a configuration file in `YAML <http://llvm.org/docs/YamlIO.html#introduction-to-yaml>`_ format.
+The Clang Static Analyzer uses taint analysis to detect injection vulnerability related issues in code.
+The backbone of taint analysis in the Clang SA is the ``TaintPropagation`` modeling checker.
```

Comment:
It is a bit strange that the `TaintPropagation` checker is hidden from the user and e.g. it doesn't have a documentation, but it still mentioned on this page and it still appears in the name of `alpha.security.taint.TaintPropagation:Config`.

I don't think that this is a blocking issue for this commit, but I think that it would be important to resolve this in the future. I can imagine two main solutions:
- We double down on the idea that "modeling checkers are hidden internal parts of the analyzer engine", eliminate all references to `TaintPropagation` and convert `alpha.security.taint.TaintPropagation:Config` into an option _of the analyzer itself_.
- We document the modeling checkers (probably in a separate section, as the alpha and debug checkers are already separate sections within the checker list). This would allow us to have modeling checkers that can be customized/configured or even disabled by the user.

@haoNoQ @Xazax-hun @steakhal What do you think about this? Which direction would be better?

Reply:
Personally, I'd prefer accepting modeling checkers as a thing and have configs for them. I'd rather not make this as an option to the analyzer itself.